### PR TITLE
Remove additionalEnv. Treat execOptions.env as always 'in addition to…

### PR DESCRIPTION
**Pull request by mattgodbolt**
… base env'. Specifically only pass those env vars through sandboxing.

Based on conversations with @apmorton and to make env vars more useful in execOptions, while keeping the flavour of "no, only the specific vars you pass will make it into the sandbox env"

See #2569

### CODE REVIEW

File: lib/base-compiler.js [hunk 1]
```diff
@@ -267,7 +267,7 @@ export class BaseCompiler {
                 timeoutMs: timeoutMs,
                 ldPath: _.union(this.compiler.ldPath, executeParameters.ldPath).join(':'),
                 input: executeParameters.stdin,
-                additionalEnv: executeParameters.additionalEnv,
+                env: executeParameters.env,
```
Comment (mattgodbolt) [inline lib/base-compiler.js:270]: Tested locally, as this worried me at first. Using firejail and this change, compiling and executing:

```cpp
#include <cstdio>

int main(int , const char *[], const char *envp[]) { 
    while (*envp) {
        printf("%s\n", *envp++);
    }
}
```

yielded
```
LD_LIBRARY_PATH=/opt/compiler-explorer/gcc-snapshot/lib:/opt/compiler-explorer/gcc-snapshot/lib32:/opt/compiler-explorer/gcc-snapshot/lib64
ASAN_OPTIONS=color=always
UBSAN_OPTIONS=color=always
MSAN_OPTIONS=color=always
LSAN_OPTIONS=color=always
```

File: lib/tooling/pvs-studio-tool.js [hunk 2]
```diff
@@ -150,7 +150,6 @@ export class PvsStudioTool extends BaseTool {
 
     getDefaultExecOptions() {
         const execOptions = super.getDefaultExecOptions();
-        execOptions.env = process.env;
         execOptions.env.PATH = process.env.PATH + ':/opt/compiler-explorer/pvs-studio-latest/bin';
```
Comment (partouf) [inline lib/tooling/pvs-studio-tool.js:153]: .env is undefined here, should be first initialized as `{}`

File: lib/exec.js [hunk 3]
```diff
@@ -24,9 +24,9 @@
 
 import child_process from 'child_process';
 import fs from 'fs';
-import path from 'path';
 
 import Graceful from 'node-graceful';
+import path from 'path';
```
Comment (apmorton) [inline lib/exec.js:29]: not sure why this was moved, but it belongs in the 'builtin' import group above.

File: lib/tooling/pvs-studio-tool.js (hunk 2 above)
Comment (apmorton) [reply lib/tooling/pvs-studio-tool.js:153]: getDefaultExecOptions will always initialize env

File: lib/base-compiler.js [hunk 4]
```diff
@@ -787,7 +787,8 @@ export class BaseCompiler {
     runExecutable(executable, executeParameters) {
         const maxExecOutputSize = this.env.ceProps('max-executable-output-size', 32 * 1024);
         // Hardcoded fix for #2339. Ideally I'd have a config option for this, but for now this is plenty good enough.
-        executeParameters.additionalEnv = {
+        executeParameters.env = {
+            ...executeParameters.env,
```
Comment (apmorton) [inline lib/base-compiler.js:791]: maybe the span should come after? so that executeParameters.env can override these hard-coded values?

**Comment by partouf** [comment]
https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/tooling/base-tool.js#L70 is different from the base-compiler implementation

File: lib/exec.js (hunk 3 above)
Comment (mattgodbolt) [reply lib/exec.js:29]: Argh stupid autoformat

File: lib/tooling/pvs-studio-tool.js (hunk 2 above)
Comment (mattgodbolt) [reply lib/tooling/pvs-studio-tool.js:153]: @partouf 's right - fixing

Comment (apmorton) [inline lib/tooling/pvs-studio-tool.js:153]: this should be:
```
execOptions.env = {...execOptions.env, PATH: process.env.PATH + ':/opt/compiler-explorer/pvs-studio-latest/bin'};
```

File: lib/base-compiler.js (hunk 4 above)
Comment (mattgodbolt) [reply lib/base-compiler.js:791]: Works for me, yup

**Comment by mattgodbolt** [comment]
I _think_ I got everything, please take another look. thanks both

File: lib/tooling/pvs-studio-tool.js [hunk 5]
```diff
@@ -33,7 +33,9 @@ import * as utils from '../utils';
 import { BaseTool } from './base-tool';
 
 export class PvsStudioTool extends BaseTool {
-    static get key() { return 'pvs-studio-tool'; }
+    static get key() {
```
Comment (apmorton) [inline lib/tooling/pvs-studio-tool.js:36]: your IDE is leaking

**Review by apmorton — APPROVED** [review]

File: lib/tooling/pvs-studio-tool.js (hunk 5 above)
Comment (mattgodbolt) [reply lib/tooling/pvs-studio-tool.js:36]: Yeah this one I don't mind /too/ much. But still; grumble, it should use the same settings as everything else.

**Pull request merged**